integration-tests/rest/src/main/java/org/apache/camel/quarkus/component/rest/it/RestResource.java

```diff
@@ -68,4 +68,26 @@ public String restProducer(@QueryParam("port") int port) {
         return producerTemplate.requestBodyAndHeaders(
                 "rest:get:/rest/template/{messageStart}/{messageEnd}?host=localhost:" + port, null, headers, String.class);
     }
+
+    @Path("/producer/binding/mode/json")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Person restProducerBindingModeJson(@QueryParam("port") int port) {
+        String query = "rest:get:/rest/binding/json/producer" +
+                "?bindingMode=json" +
+                "&outType=org.apache.camel.quarkus.component.rest.it.Person" +
+                "&host=localhost:" + port;
```

Comment:
Looking at this, I am am why we need a JAX-RS endpoint to test the rest component? This is perhaps a way to pass `bindingMode=json`, but it it can also be passed via REST DSL, which we already did before this PR, see https://github.com/apache/camel-quarkus/pull/3052/files#diff-58cbc36531a538cc13ef975a1889ac4ec3b28c54786b5896118916846558a592R64-R71 for JSON and https://github.com/apache/camel-quarkus/pull/3052/files#diff-58cbc36531a538cc13ef975a1889ac4ec3b28c54786b5896118916846558a592R78-R85 for XML. 
Is this PR adding anything new the above two exist?

Reply:
Hello @ppalaga,
the above two test only Consumer while addition in this PR covers Producer. As you can see within this test coverage result https://fuse-next-jenkins-csb-fuse-qe.apps.ocp4.prod.psi.redhat.com/job/Integration.next/job/camel-quarkus/job/code-coverage/43/Camel_20JaCoCo_20report/org.apache.camel.component.rest/RestProducer.java.html#L272 
method `createBindingProcessor()` is almost entirely untested. With those changes the coverage increase greatly.

Reply:
Thanks for explaining, @vhais!